spec/models/game_spec.rb

```diff
@@ -46,7 +46,19 @@
         expect(Piece.where(type: piece, color:'white').count).to eq(number)
         expect(Piece.where(type: piece, color:'black').count).to eq(number)
       end
-      
+
+    end
+  end
+
+  describe '.check' do
+
```

Comment:
[check_method_test_sample.txt](https://github.com/rosalinekarr/WarAndPiece/files/1490391/check_method_test_sample.txt)
We will need more tests to make sure this method is working properly.  Unfortunately, it's not enough to provide test coverage for a generic piece since each piece has their own unique valid_move? method.  I'll provide some code you can use to get a better idea for testing and to start off with. I'll let you finish adding the other pieces.

The crucial part that is missing from your test is your action part which in this case is **moving to the position and calling the check? method**

```
      it 'is checked by Pawn' do
        @pawn = FactoryGirl.create(:pawn,  file: 3, rank: 1, game: @game, color: :white_player_id)

        @pawn.move_to!(3, 3)
        result = @game.check?(@pawn)

        expect(result).to be true
      end
```
Notice there are 3 parts to this: arranging your data, acting on that data, and that making your assertion.  These are known as the the Three A's test pattern: Arrange, act, assert.

Also, you may omit result= and instead opt for expect(@game.check?(@pawn).to be true but I find it a lot more readable to use a variable in your assertion.

One other note, the pawn has a special case to it since it captures diagonally and currently there isn't any logic in place yet to handle that.  You can include it in the test but be sure to make it pending by attaching an 'x' to the it part of the test like below:

`xit 'is checked by Pawn' do,`

This will omit the test and show as pending when you run rspec